geo/src/algorithm/concave_hull.rs

```diff
@@ -0,0 +1,432 @@
+            .collect();
+        Polygon::new(
+            LineString::from(concave_hull(&aggregated, concavity)),
+            vec![],
+        )
+    }
+}
+
+impl<T> ConcaveHull<T> for MultiPoint<T>
+where
+    T: Float + RTreeNum,
+{
+    fn concave_hull(&self, concavity: T) -> Polygon<T> {
+        Polygon::new(
+            LineString::from(concave_hull(&self.0.clone(), concavity)),
+            vec![],
+        )
+    }
+}
+
+fn find_point_closest_to_line<T>(
+    interior_points_tree: &RTree<Point<T>>,
+    line: Line<T>,
+    max_dist: T,
+    edge_length: T,
+    concavity: T,
+    line_tree: &RTree<Line<T>>
+) -> Option<Point<T>>
+where
+    T: Float + RTreeNum,
+{
+    let h = max_dist + max_dist;
+    let w = line.euclidean_length() + h;
+    let two = T::add(T::one(), T::one());
+    let search_dist = T::div(T::sqrt(T::powi(w, 2) + T::powi(h, 2)), two);
+    let mut candidates = interior_points_tree
+        .locate_within_distance(line.centroid(), search_dist)
+        .peekable();
+    let peek = candidates.peek();
+    match peek {
+        None => None,
+        Some(&point) => {
+            let closest_point = candidates.fold(*point, |acc, candidate| {
+                if line.euclidean_distance(&acc) > line.euclidean_distance(candidate) {
+                    *candidate
+                } else {
+                    acc
+                }
+            });
+            let mut edges_nearby_point = line_tree
+                .locate_within_distance(closest_point, search_dist)
+                .peekable();
+            let peeked_edge = edges_nearby_point.peek();
+            let closest_edge_option = match peeked_edge{
+                None => None,
+                Some(&edge) => Some(edges_nearby_point.fold(*edge, |acc, candidate| {
+                    if closest_point.euclidean_distance(&acc) > closest_point.euclidean_distance(candidate) {
+                        *candidate
+                    } else {
+                        acc
+                    }
+                }))
+            };
+            let decision_distance = partial_min(
+                closest_point.euclidean_distance(&line.start_point()),
+                closest_point.euclidean_distance(&line.end_point()),
+            );
+            if let Some(closest_edge) = closest_edge_option {
+               let far_enough = edge_length / decision_distance > concavity;
+               let are_edges_equal = closest_edge == line;
+               if far_enough && are_edges_equal {
+                   Some(closest_point)
+               } else {
+                   None
+               }
+            }else {
+               None
+            }
+        }
+    }
+}
+
+// This takes significant inspiration from:
+// https://github.com/mapbox/concaveman/blob/54838e1/index.js#L11
+fn concave_hull<T>(points: &Vec<Point<T>>, concavity: T) -> Vec<Point<T>>
+where
+    T: Float + RTreeNum,
+{
+    let multipoint: MultiPoint<T> = points.clone().into();
```

Comment:
Can this clone be avoided?  Note that the trait impl for geometry types already clones the individual coordinates, when populating the `Vec`.  I understand that we do not implement `ConvexHull` on a slice of coordinates (may be we should).  As a temporary soln., you could go for a directly use the `qhull::quick_hull` algorithm.  It just needs a mutable slice of Coordinates, which you can get from a vector of coords.

May be the function could read: `fn concave_hull<T>(points: &[Coordinate<T>], ...)` just to make it more flexible.